core/src/libraries/BalanceSheet.sol

```diff
@@ -119,4 +119,18 @@ library BalanceSheet {
             }
         }
     }
+
+    /// @dev Equivalent to log_{1.0001}(1e18 / cf) / 12 assuming `MANIPULATION_THRESHOLD_DIVISOR = 24`
+    function _manipulationThreshold(uint256 cf) private pure returns (uint24) {
+        return uint24(-TickMath.getTickAtSqrtRatio(uint160(cf)) - 501937) / MANIPULATION_THRESHOLD_DIVISOR;
```

Comment:
What is 501937?

Reply:
Our goal is to calculate $log_{1.0001}$ of the reciprocal of the collateral factor. But instead of writing a new function that calculates $log_{1.0001}$ directly, we want to use the existing `TickMath.getTickAtSqrtRatio(x)`.

Let $F(x)$ represent `TickMath.getTickAtSqrtRatio(x)`

$F(2^{96} * \sqrt{x}) = log_{1.0001}x$

Let $y = \frac{x^2}{2^{192}}$. Then we have:

$F(2^{96} * \sqrt{y}) = log_{1.0001}y$
$F(2^{96} * \sqrt{\frac{x^2}{2^{192}}}) = log_{1.0001}(\frac{x^2}{2^{192}})$
$F(2^{96} * \frac{x}{2^{96}}) = 2log_{1.0001}(x) - 192log_{1.0001}(2)$
$F(x) = 2log_{1.0001}(x) - 192log_{1.0001}(2)$

Rearranging, we get:

$log_{1.0001}(x) = \frac{1}{2} \cdot (F(x) + 192log_{1.0001}(2))$
$log_{1.0001}(x) = \frac{1}{2} \cdot (F(x) + 1330909.128)$

Then we deal with the extra factor of $10^{18}$ in our collateral factor.

$log_{1.0001}(c * 10^{18}) = log_{1.0001}(c) + 18log_{1.0001}(10)$
$log_{1.0001}(c * 10^{18}) = log_{1.0001}(c) + 414486$
$log_{1.0001}(c) = log_{1.0001}(c * 10^{18}) - 414486$

Plugging in the previous equation:

$log_{1.0001}(c) = \frac{1}{2} \cdot (F(c * 10^{18}) + 1330909.128) - 414486.04$
$log_{1.0001}(c) = \frac{1}{2} \cdot (F(c * 10^{18}) + 1330909.128 - 828972.07)$
$log_{1.0001}(c) = \frac{1}{2} \cdot (F(c * 10^{18}) + 501937.06)$

Finally, taking the reciprocal is the same as multiplying by -1 in log space, so:

$log_{1.0001}(\frac{1}{c}) = \frac{1}{2} \cdot (-F(c * 10^{18}) - 501937.06)$

Translating to code and ignoring data types, this would be:

```Solidity
(-TickMath.getTickAtSqrtRatio(cf) - 501937) / 2;
```

This is essentially the base case for the manipulation threshold, so you can see the `MANIPULATION_THRESHOLD_DIVISOR` that I use in the actual code is just to scale things down and make our heuristic more sensitive.

Reply:
TLDR:

$`501937 = 192log_{1.0001}(2) - 2*18*log_{1.0001}(10) = \frac{192log(2) - 36log(10)}{log(1.0001)}`$